frontend/src/features/books/EditBookForm.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+import { useParams, useNavigate } from 'react-router-dom'
+import { useEffect, useState } from 'react'
+import { API_URL } from '../../constants'
+
+function EditBookForm() {
+    const [book, setBook] = useState(null)
+    const [title, setTitle] = useState('')
+    const [author, setAuthor] = useState('')
+    const [description, setDescription] = useState('')
+    const { id } = useParams()
+    const [, setLoading] = useState(true)
+    const [, setError] = useState(null)
+    const navigate = useNavigate()
+
+    useEffect(() => {
+        const fetchCurrentBook = async () => {
+            try {
+                const response = await fetch(`${API_URL}/${id}`)
+                if (response.ok) {
+                    const json = await response.json()
+                    setBook(json)
+                    setTitle(json.title)
+                    setAuthor(json.author)
+                    setDescription(json.description)
+                } else {
+                    throw response
+                }
+            } catch (e) {
+                console.log("an error occurred")
+            } finally {
+                setLoading(false)
+            }
+        }
+        fetchCurrentBook()
+    }, [id])
+
+    const handleSubmit = async (e) => {
+        e.preventDefault()
+        const updatedBook = {
+            ...book,
+            title,
+            author,
+            description
+        }
+        try {
+            const response = await fetch(`${API_URL}/${id}`, {
+                method: "PUT",
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                body: JSON.stringify(updatedBook)
+            })
+            if (response.ok) {
+                const json = await response.json()
```

Comment:
@nshelokar-vf, what is `json`?
Can we have better name here?

Reply:
Sure. I will make sure from next time.